synapse/lib/hiveauth.py

```diff
@@ -38,6 +38,42 @@ def textFromRule(rule):
         text = '!' + text
     return text
 
+def formatAllowedReason(resp):
```

Comment:
Any reason we can't just embed `mesg` into the reason dict?

Reply:
We definitely could; I was trying to keep the amount of cached information in this low since my original use case for making this API function didn't need a human friendly message.